src/rules/noMagicNumbersRule.ts

```diff
@@ -101,11 +135,29 @@ class NoMagicNumbersWalker extends Lint.AbstractWalker<number[]> {
         return ts.forEachChild(sourceFile, cb);
     }
 
-    private checkNumericLiteral(node: ts.Node, num: string) {
+    private isAllowedNumber(num: string): boolean {
         /* Using Object.is() to differentiate between pos/neg zero */
+        const compareNumbers = (allowedNum: number) => Object.is(allowedNum, parseFloat(num));
+        return this.options.some(option => {
```

Comment:
> `this.options.some`

Now that the rule options are coming in as a more complex format, it's getting difficult to read through the rule and see how the options are meant to be used. Proposal: let's switch the responsibility of parsing options from the walker to the rule itself, such that the rule parses the options into an object of a known format & passes that object to the walker. Say, a `Set` of allowed numbers and a boolean for the new option?

`arrowReturnShorthandRule.ts` is a good simple case of this strategy, where the object passed to the walker just has the `multiline` property. `banRule.ts` has a complex object so it uses a `parseOptions` function.

Reply:
Ok, that  makes sense!